rotkehlchen/chain/ethereum/utils.py

```diff
@@ -166,3 +174,38 @@ def should_update_protocol_cache(cache_key: GeneralCacheType, *args: str) -> boo
             key_parts=[cache_key, *args],
         )
     return ts_now() - last_update_ts >= ETH_PROTOCOLS_CACHE_REFRESH
+
+
+def try_download_ens_avatar(
+        eth_inquirer: 'EthereumInquirer',
+        avatars_dir: Path,
+        ens_name: str,
+) -> None:
+    """
+    Handles ens avatar downloading.
+    1. Checks whether given ens name has avatar set
+    2. If it does -- downloads the avatar and saves it in `avatars_dir`
+    3. Updates last avatar check timestamp for the given ens name
+
+    May raise:
+    - RemoteError if failed to query chain
+    - RequestsError if got avatar url, but failed to download it
+    """
+    dbens = DBEns(eth_inquirer.database)
+    avatar_url = eth_inquirer.ens_avatar_lookup(ens_name)
+    dbens.set_last_avatar_update(ens_name=ens_name, timestamp=ts_now())
+    if avatar_url is None:
+        return  # Avatar is not set
+
+    try:
+        avatar = requests.get(avatar_url, timeout=10).content
+    except RequestException as e:
+        log.info(
+            f'Got {str(e)} during querying ens avatar for {ens_name}. '
+            f'Considering the url broken and the avatar non-existent.',
```

Comment:
Short and to the point. The logs are for us to read and we will know when that happened.

So here if this happens avatar won't be attempted to be fetched for another day, right?

```suggestion
            f'Got error {str(e)} during querying ens avatar for {ens_name}. '
```

Reply:
> So here if this happens avatar won't be attempted to be fetched for another day, right?

Yes. Since it can easily be not a temporary network error, but a broken link set in the ens app I think that retrying to download the avatar every time is too much.